Kitodo/src/main/java/org/kitodo/production/services/index/IndexingService.java

```diff
@@ -222,51 +226,39 @@ public void countDatabaseObjects() throws DAOException {
         }
     }
 
-    private void prepareIndexWorker() throws DAOException {
-
-        int indexLimit = ConfigCore.getIntParameterOrDefaultValue(ParameterCore.ELASTICSEARCH_INDEXLIMIT);
-        for (ObjectType objectType : ObjectType.values()) {
-            List<IndexWorker> indexWorkerList = new ArrayList<>();
-
-            int databaseObjectsSize = getNumberOfDatabaseObjects(objectType);
-            if (databaseObjectsSize > indexLimit) {
-                int start = 0;
-
-                while (start < databaseObjectsSize) {
-                    indexWorkerList.add(new IndexWorker(searchServices.get(objectType), start));
-                    start += indexLimit;
-                }
-            } else {
-                indexWorkerList.add(new IndexWorker(searchServices.get(objectType)));
-            }
-
-            indexWorkers.put(objectType, indexWorkerList);
-        }
-    }
-
     /**
-     * Index all objects of given type 'objectType'.
+     * Manage indexing for a given object type.
+     * 
+     * <p>This method is executed in the `IndexManagementThread`.</p>
      *
      * @param type
      *            type objects that get indexed
      */
-    public void startIndexing(ObjectType type, PushContext pushContext) throws DataException, CustomResponseException {
+    public void runIndexing(ObjectType type, PushContext pushContext, boolean indexAllObjects) 
+            throws DataException, CustomResponseException, DAOException {
         SearchService searchService = searchServices.get(type);
         int indexLimit = ConfigCore.getIntParameterOrDefaultValue(ParameterCore.ELASTICSEARCH_INDEXLIMIT);
-        if (countDatabaseObjects.get(type) > 0) {
-            List<IndexWorker> indexWorkerList = indexWorkers.get(type);
-            Long amountInIndex = searchService.count();
-            long indexBatches = 0L;
 
-            while (indexBatches < amountInIndex) {
-                searchService.removeLooseIndexData(searchService.findAllIDs(indexBatches, indexLimit));
-                indexBatches += indexLimit;
+        if (countDatabaseObjects.get(type) > 0) {
+            if (indexAllObjects) {
+                // only check for loose index data when indexing full objects
+                Long amountInIndex = searchService.count();
+                long offset = 0L;
+
+                // remove documents in index that are no longer available in database
+                // by iterating over all indexed documents in elastic search
+                try {
+                    while (offset < amountInIndex) {
+                        // TODO: actually iterate over all elastic search documents
```

Comment:
When will this TODO be solved? Is there is already changed ready or will this be fixed in a middle / far future away change?

Reply:
This code was not changed by this pull request. I just added a comment that this is a problem. There is a new issue #5379 for this.

Reply:
Thanks, then will this TODO / Issue not got lost.